### PR TITLE
Updated tools swagger json converter to generate an admin api json

### PR DESCRIPTION
**Pull request by harageth**
# Resolves

https://jira.rax.io/browse/SALUS-990

# What

Generate an admin swagger json as well as clean up some of the generation of html

# How

It detects the admin and the public api endpoints and generates swagger json for each one. 

# How to test

Run the maven `deploy` command on this repo.
Run the maven test command on monitor management from the branch with the same name. It will generate the public and admin swagger json.

# Why

I thought about altering the templates that were provided to us so it didn't go and download the js each time but that seemed like a bad idea. We already have a different tool for generating the html that shows up in google cloud storage, so it seemed best to make this tool only generate swagger json and leave html generation to the other tool.

# TODO

Update configs in resource, event-engine, agent-catalogue, and policy management

### CODE REVIEW

**Comment by GeorgeJahad** [comment]
The changes seem fine but I'm not quite clear on the html generation that has been removed.  That is now being done by a different tool?  does that tool need to know about the new json files generated here?

**Comment by harageth** [comment]
> The changes seem fine but I'm not quite clear on the html generation that has been removed. That is now being done by a different tool? does that tool need to know about the new json files generated here?

It does. Its being handle here and will be configured as such in each other repository that is generating API docs. https://github.com/racker/salus-telemetry-monitor-management/pull/202/files#diff-600376dffeb79835ede4a0b285078036R327-R336

File: src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java [hunk 1]
```diff
@@ -32,16 +49,20 @@
     public static void main(String[] args) throws Exception {
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:49]: Is the description above still accurate?

Comment (harageth) [reply src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:49]: Oh yeah I need to change that now too. I altered the number of args.

File: src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java [hunk 2]
```diff
@@ -61,40 +82,30 @@ public static void main(String[] args) throws Exception {
                 }
             });*/
             if(containsTenant) {
-                temp.put(newKey, elt.getValue());
+              publicTemp.put(newKey, elt.getValue());
+            }else if(containsAdmin){
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:86]: I wonder if just an `else` would have been safer here? Spot checking, it does appear we always prefix admin API paths with `/admin`; however, simply assuming non-tenant APIs are admin APIs would avoid any escapes.

File: src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java [hunk 3]
```diff
@@ -20,28 +37,32 @@
...
-        Map<String, JsonNode> temp = new HashMap<>();
+        ObjectNode publicRoot = (ObjectNode)mapper.readTree(content);
+        ObjectNode adminRoot = (ObjectNode)mapper.readTree(content);
+        Map<String, JsonNode> publicTemp = new HashMap<>();
+        Map<String, JsonNode> adminTemp = new HashMap<>();
 
         String newKey;
         boolean containsTenant;
-        for (Iterator<Map.Entry<String, JsonNode>> it = root.get("paths").fields(); it.hasNext(); ) {
+        boolean containsAdmin;
```
[the remark this answers is not in the document]
Comment (harageth) [reply src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:59]: Yeah not sure why I declared them outside the loop like that when I originally wrote it. Looks like the same is true for newKey

File: src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java [hunk 4]
```diff
@@ -54,14 +54,11 @@ public static void main(String[] args) throws Exception {
...
-        boolean containsTenant;
-        boolean containsAdmin;
         for (Iterator<Map.Entry<String, JsonNode>> it = publicRoot.get("paths").fields(); it.hasNext(); ) {
             Map.Entry<String, JsonNode> elt = it.next();
-            newKey = elt.getKey();
-            containsTenant = newKey.contains("tenant");
-            containsAdmin = newKey.contains("admin");
+            String newKey = elt.getKey();
+            boolean containsTenant = newKey.contains("tenant");
+            boolean containsAdmin = newKey.contains("admin");
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:61]: Perhaps can remove `containsAdmin` now.

**Pull request merged**